### PR TITLE
Nitpicking

### PR DESCRIPTION
**Pull request by jlnr**
I've tried to limit this PR to the things that I feel are definitely better: Fixing typos, more consistent formatting, a bit of consistency here and there, not being too clever with the `^` operator :)

I'm not sure if making the string helpers class methods was the right call, though. Feel free to just cherry-pick some bits here and there...

Some other suggestions:
1. I don't like how `setTimeZone:` magically sets `self.calendar.timeZone` as well, same for the locale. Why not just use the calendar's `timeZone`/`locale`? That way you only have to encode a single object, and data cannot get out of sync. As a compromise, you could offer the `timeZone`/`locale` properties but route them through to the `calendar`'s properties (no backing ivar)
2. I'd start the instance variables with an _underscore to be consistent with auto-generated `@property` ivars
3. Not sure what the _Pods symlink is doing in the root directory :)
4. `.gitkeep` in `Classes` doesn't serve a purpose since the directory is non-empty
5. I wouldn't use CocoaPods for the example, just add the files in `Classes` manually and you can keep ~10 files out of git (also - I see an warning that frameworks only work on iOS 8 - do you use frameworks but set an iOS 7 deployment target?)
6. I think the backwards-compatible constants for time units are over-engineered; can you even submit iOS apps with the old SDKs that don't support the new constants? => I'd just use the new constants instead...
7. Should it default to `NSLocale.currentLocale` (as it does right now) or to `NSBundle.mainBundle.preferredLocalizations.firstObject` (or maybe not the `mainBundle`, but the resource bundle)? Honest question, I have no idea.
8. File organisation nitpick: I like having the `Classes` folder in the root of the git repository. Flatten all the things 💪🏽
9. From a quick glance, the `NSCoding` impl only seems to encode/decode all the properties, not the ivars?

### CODE REVIEW

**Comment by sebastianludwig** [comment]
1. This came to be because there's also an internal `NSDateFormatter` and its time zone and locale need to be in sync with the one in the calendar. Setting the `calendar` property on `NSDateFormatter` doesn't make it use the calendars time zone and locale. Also the calendar may come without locale (is's optional in the Swift interface). So I though I'd be a good idea to kind of copy the `NSDateFormatter` pattern and let the `SLCDF` be in charge of the locale and time zone. OTOH, sleek interfaces ftw, the calendar could rule them all. If no locale is set, well then none is set. ATM it's also possible to assign `nil`.
2. Till now I only start property backing ivars with an underscore to explicitly distinguish between them and plain ivars. This way it's easier to decide if one should use the accessor methods or the ivar directly while writing code. Not a good idea you reckon? Like prefixing a variable with its type in good ol' VB?
3. No clue either. I just typed `pod lib create`. I guess it's there for a reason.
4. :heavy_check_mark: 
5. I'll try. Again, this is just what `pod lib create` did. Same for the frameworks and iOS 7. Maybe it's not such a good idea to use it after all...
6. Just took it from [mattt/FormatterKit] tbh, but agreed.
7. Doesn't matter much, if the calendar becomes the source of the locale (which is another +1 for that approach)
8. `pod lib create`... I'll try moving things around. If anything breaks and it takes longer than 3 minutes to fix, I'll leave it as is :-)
9. Whops. Good catch!

Thank you very much for the thorough review!

File: Pod/Classes/SLConditionalDateFormatter.m [hunk 1]
```diff
@@ -141,9 +141,10 @@ - (BOOL)isInFuture
 @end
 
 
-static inline NSComparisonResult NSCalendarUnitCompareSignificance(NSCalendarUnit a, NSCalendarUnit b)
```
Comment (sebastianludwig) [inline Pod/Classes/SLConditionalDateFormatter.m:144]: y u no inline?

Comment (jlnr) [inline Pod/Classes/SLConditionalDateFormatter.m:144]: The rules for 'inline' in C are a little more complicated than in C++, and even in C++ I'd trust the compiler to make the right choice on its own.
I only use 'inline' to be able to define functions in header files and include them in several .cpp files without linker errors (which only works in C++), not as an optimisation hint. Less characters = more clarity :)

**Comment by jlnr** [comment]
1. That makes sense, I didn't think of the internal `dateFormatter`.
2. I'm not sure if there is a convention for it, but I like being able to look at a variable and see that it is an _ivar (like Ruby's `@ivars`) vs. a local variable or parameter (no underscore).
3. Wait, why have I never heard of this command... ^^"

**Pull request merged**